pkg/common/x509util/serialnumber.go

```diff
@@ -1,23 +1,28 @@
 package x509util
 
 import (
-	"context"
+	"crypto/rand"
+	"fmt"
+	"math"
 	"math/big"
-	"sync/atomic"
 )
 
-type SerialNumber interface {
-	NextNumber(context.Context) (*big.Int, error)
-}
-
-type serialNumber struct {
-	next int64
-}
+// NewSerialNumber creates a random certificate serial number according to CA/Browser forum spec
+// Section 7.1:
+//   "Effective September 30, 2016, CAs SHALL generate non-sequential Certificate serial numbers greater than
+//   zero (0) containing at least 64 bits of output from a CSPRNG"
+func NewSerialNumber() (*big.Int, error) {
+	// Get MaxUint64 as big.Int
+	maxUInt64 := new(big.Int)
+	maxUInt64 = maxUInt64.SetUint64(math.MaxUint64)
```

Comment:
nit: use package level unexported vars for these "const" values
```
var (
    maxUint64 = new(big.Int).SetUint64(math.MaxUint64)
    one = big.NewInt(1)
)
```
This is more for readability than performance (it keeps NewSerialNumber() small)